### ISSUE TITLE
Layer controls relationships

### ISSUE THREAD

**Issue by liaprins-czi** [napari/product-heuristics-2020#19]
The actions shown per layer type, as well as the parameters are not consistently communicated in terms of their relationships and dependencies: For example, sometimes a parameter (slider, etc) only applies to selected things in the layer, other times to the layer as a whole. Also, some of the actions are dependent upon other actions having been done first, or are very related to each other relative to other actions, but this is not clear. 

The points' layer "x" button, the shapes' layer "remove vertex" and "add vertex" buttons (and probably others) are clickable but nothing happens. Takes a lot of trial and error to figure out why. Might also be relevant for "Display text" and "n-dim" checkboxes.

Hard to keep track of the opacity slider and blending mode together, as they affect each other/are related.
Unclear (before using) what the opacity slider in the labels layer applies to when eraser action is selected: the eraser's opacity or the background.

When no layer is selected, the "layer control" window is blank. This can lead to to confusion about intended user action and intention of space.

The order that an individual layer's actions appear in are confusing: For example, the shapes layer's action buttons first show the remove vertex and add vertex buttons, but they cannot be used until a shape is first created, but the shape creation buttons are last. Also, the buttons appear to be stacked from bottom to top, right to left (both backwards of standard) leaving a strange gap in upper left.

The "- 1 +" widget is the wrong choice for toggling between label colors, as it traditionally signifies increased quantity of a single entity, not multiple as used here. It also makes it harder to move back and forth between label colors (ex: picking label color 5, then going back to label color 1 requires four clicks).
__________________________________________________________
**Recommendation**

Overhaul UI to better reflect relationship between actions, between parameters, and between actions and parameters. Specifically:

- Actions that are not possible (due to no points or shapes existing on the layer yet, or any other dependency) should be disabled, until it is possible to use them.
- Points layer's opacity slider and blending mode are related/dependent upon each other, so should be more visually and interactively tied together.
- Consider blending mode applied to layer as a whole (like it is for image layers). Otherwise should be disabled unless something it can be applied to is selected.
- Consider adding starter text to empty box to orient users, or making it so a layer is always selected.
- Stack buttons from left to right, then overflow to next line below. Start with the buttons users would need to use first/most in terms of task completion. Buttons and drop downs should be ordered by priority of action (and grouped with similar functions, if not already done).
- Choose different widget (for label layer color selection): one that displays the color on the window once it is selected and used, for users to see and compare, and toggle back to as needed.

Moderate issue
●●●   High design work
▼▼▼   High development work

When no layer is selected, the "layer control" window is blank:
<img width="256" alt="layercontrolempty" src="https://user-images.githubusercontent.com/53277128/95918708-2180a580-0d61-11eb-933f-0c3ad16e5795.png">

The buttons appear to be stacked from bottom to top, right to left (both backwards of standard) leaving a strange gap in upper left:
<img width="265" alt="buttonposition" src="https://user-images.githubusercontent.com/53277128/95920502-66f2a200-0d64-11eb-9860-585858061f49.png">

The "- 1 +" widget is the wrong choice for toggling between label colors:
<img width="258" alt="colorpick" src="https://user-images.githubusercontent.com/53277128/95920589-85589d80-0d64-11eb-9f23-ac8db5965f54.png">

**Comment by LCObus** [comment]
While pre-design/engineering UXR is not required, this will be part of the usability test post change, to evaluate user mental models on relationships between layers and panel functionality.

**Comment by goanpeca** [comment]
@LCObus is there additional context or work that needs to be done before this can be tackled?

Thanks :)

**Comment by LCObus** [comment]
Thanks for checking @goanpeca , but I think Lia laid it out appropriately to tackle!

**Comment by goanpeca** [comment]
Cool, this is something else we could work on @ppwadhwa

**Comment by liaprins-czi** [comment]
@goanpeca @ppwadhwa Actually this one will take a lot of design effort; I will not really be able to focus on it until I am finished with the first milestone of the Hub work. However, there may be a couple specific aspects I called out which could be fixed now as described, i.e. button stacking alignment.

**Comment by goanpeca** [comment]
Sounds good @liaprins-czi!

**Comment by liaprins-czi** [comment]
Related: https://github.com/napari/napari/issues/3104

**Comment by liaprins-czi** [comment]
Related: https://github.com/napari/napari/issues/2337

**Comment by liaprins-czi** [comment]
Related: [Inconsistent behaviour when adding points (some options affect all points in layer, some don't) #702](https://github.com/napari/napari/issues/702)
